game-core/src/main/java/games/strategy/triplea/delegate/battle/BattleState.java

```diff
@@ -92,6 +92,16 @@ public boolean isFirstRound() {
 
   Collection<Unit> removeNonCombatants(Side side);
 
+  /**
+   * Mark the units that will be dying
+   *
+   * <p>Units that are only damaged should not be passed into this method.
+   *
+   * @param casualties units that are dying
+   * @param side the side the unit are on
+   */
+  void markCasualties(Collection<Unit> casualties, Side side);
```

Comment:
Which would you prefer, the javadoc on the interface, the implementation, or both?

Reply:
IMO interface to explain the general contract that is expected to be provided by any implementation.